tests/2d_examples/test_2d_collision/src/collision.cpp

```diff
@@ -239,8 +241,10 @@ int main(int ac, char* av[])
 	solid_dynamics::StressRelaxationSecondHalf damping_ball_stress_relaxation_second_half(damping_ball_inner);
 	/** Algorithms for solid-solid contact. */
 	solid_dynamics::ContactDensitySummation free_ball_update_contact_density(free_ball_contact);
+	solid_dynamics::ContactDensitySummation wall_free_ball_update_contact_density(wall_free_ball_contact);
```

Comment:
Dear @Xiangyu-Hu  in order to have the contact_density being calculated for the second contacting body (in a contact pair), one must calculate the contact_density also seperately for each contacting body, with the associated contact relation. Otherwise, the contact_density for other contacting bodies will be 0.